app/bot/openai.go

```diff
@@ -18,38 +19,73 @@ type OpenAIClient interface {
 	CreateChatCompletion(context.Context, openai.ChatCompletionRequest) (openai.ChatCompletionResponse, error)
 }
 
+// OpenAIConfig contains parameters for OpenAI bot
+type OpenAIConfig struct {
+	AuthToken       string
+	MaxTokens       int
+	Prompt          string
+	HistorySize     int
+	HistoryRandBase int64
+}
+
 // OpenAI bot, returns responses from ChatGPT via OpenAI API
 type OpenAI struct {
-	authToken string
-	client    OpenAIClient
-	maxTokens int
-	prompt    string
+	client OpenAIClient
+
+	config    OpenAIConfig
 	superUser SuperUser
 
+	history LimitedMessageHistory
+	rand    func(n int64) int64 // tests may change it
+
 	nowFn  func() time.Time // for testing
 	lastDT time.Time
 }
 
 var maxMsgLen = 14000
 
-// NewOpenAI makes a bot for ChatGPT
-// maxTokens is hard limit for the number of tokens in the response
-// https://platform.openai.com/docs/api-reference/chat/create#chat/create-max_tokens
-func NewOpenAI(authToken string, maxTokens int, prompt string, httpClient *http.Client, superUser SuperUser) *OpenAI {
-	log.Printf("[INFO] OpenAI bot with github.com/sashabaranov/go-openai, prompt=%s, max=%d", prompt, maxTokens)
+func initOpenAIClient(authToken string, httpClient *http.Client) OpenAIClient {
 	config := openai.DefaultConfig(authToken)
 	config.HTTPClient = httpClient
 
-	client := openai.NewClientWithConfig(config)
-	return &OpenAI{authToken: authToken, client: client, maxTokens: maxTokens, prompt: prompt,
-		nowFn: time.Now, superUser: superUser}
+	return openai.NewClientWithConfig(config)
+}
+
+// NewOpenAI makes a bot for ChatGPT
+// MaxTokens is hard limit for the number of tokens in the response
+// https://platform.openai.com/docs/api-reference/chat/create#chat/create-max_tokens
+func NewOpenAI(config OpenAIConfig, httpClient *http.Client, superUser SuperUser) *OpenAI {
+	log.Printf("[INFO] OpenAI bot with github.com/sashabaranov/go-openai, Prompt=%s, max=%d",
+		config.Prompt, config.MaxTokens)
+	client := initOpenAIClient(config.AuthToken, httpClient)
```

Comment:
any particular reason why this should a a function? If you really want to wrap it to a function I'd suggest making it internal, inside NewOpenAI

Reply:
My main reason was the isolating for `config` for `OpenAI` bot and `config` for `OpenAIClient`
The renaming `config` to `params` would help with resolving this name conflict